mitodl/course-search-utils
# fix history dep

### Relevant Tickets

#58, https://github.com/mitodl/open-discussions/pull/4115

### Description (What does it do?)
This PR:
1. changes `history` from a dependency to a peer dependency.
2. Updates `prettier`, since we needed to `import type { ... } from "history"`, and `import type` is not supported by Prettier v1

See code comments.

### How should this be tested?
In Open discussions:
```sh
git checkout cc/update-course-search-utils
docker compose run --rm watch yarn install # or docker-compose up
```
Check that search still works.
